segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ExtendedS3IntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+        bookkeeper.initialize();
+
+        String endpoint = "http://127.0.0.1:" + TestUtils.getAvailableListenPort();
+
+        URI uri = URI.create(endpoint);
+
+        baseDir = Files.createTempDirectory("extendeds3_wrapper").toString();
+
+        this.configBuilder.include(ExtendedS3StorageConfig.builder()
+                                                          .with(ExtendedS3StorageConfig.BUCKET, "kanpravegatest")
+                                                          .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                          .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                          .with(ExtendedS3StorageConfig.URI, endpoint));
+    }
+
+    /**
+     * Shuts down BookKeeper and cleans up file system directory
+     */
+    @After
+    public void tearDown() throws Exception {
+        bookkeeper.close();
+        FileHelpers.deleteFileOrDirectory(new File(baseDir));
+    }
+
+    //endregion
+
+    //region StreamSegmentStoreTestBase Implementation
+
+    @Override
+    protected ServiceBuilder createBuilder(ServiceBuilderConfig builderConfig) {
+        return ServiceBuilder
+                .newInMemoryBuilder(builderConfig)
+                .withCacheFactory(setup -> new RocksDBCacheFactory(builderConfig.getConfig(RocksDBConfig::builder)))
+                .withStorageFactory(setup -> {
+                    StorageFactory f = new ExtendedS3StorageFactory(
+                            setup.getConfig(ExtendedS3StorageConfig::builder), setup.getExecutor());
+                    return new ExtendedS3ListenableStorageFactory(f, builderConfig.getConfig(ExtendedS3StorageConfig::builder), this.aclMap);
+                })
+                .withDataLogFactory(setup -> new BookKeeperLogFactory(setup.getConfig(BookKeeperConfig::builder),
+                        bookkeeper.getZkClient(), setup.getExecutor()));
+    }
+
+
+    private class ExtendedS3ListenableStorageFactory extends ListenableStorageFactory {
+
+        private final ExtendedS3StorageConfig adapterConfig;
+        private final AclMap aclMap;
+
+        public ExtendedS3ListenableStorageFactory(StorageFactory wrappedFactory, ExtendedS3StorageConfig adapterConfig, AclMap aclMap) {
+            super(wrappedFactory);
+            this.adapterConfig = adapterConfig;
+            this.aclMap = aclMap;
+        }
+
+        @Override
+        public Storage createStorageAdapter() {
+            URI uri = URI.create("http://localhost:9020");
+            S3Config s3Config = new S3Config(uri);
+
+            s3Config = s3Config.withIdentity(adapterConfig.getAccessKey()).withSecretKey(adapterConfig.getSecretKey())
+                               .withRetryEnabled(false).withInitialRetryDelay(1).withProperty("com.sun.jersey.client.property.connectTimeout", 100);
+
+            S3JerseyClient client = new S3ClientWrapper(s3Config, this.aclMap, baseDir);
```

Comment:
The wrapped factory is the production code which creates a standard S3 client. This created a test implementation of the client.